public/main.js

```diff
@@ -0,0 +1,199 @@
+    currentPlayer = playerX;
+    status.textContent = `${currentPlayer.name}'s Turn`
+    button.disabled = true;
+    gameplay();
+}
+
+function gameplay() {
+    timer = setInterval(countUp, 1000)
+    squares.forEach((square) => { square.addEventListener('click', select) })
+}
+
+function select() {
+    let square = event.target;
+    let i = squares.indexOf(square);
+    if (squares.includes(square) && !usedMoves.includes(parseInt(square.id))) {
+        square.textContent = currentPlayer.symbol;
+        currentPlayer.moves.push(parseInt(squares.slice(i, i + 1)[0].id));
+        usedMoves.push(parseInt(squares.slice(i, i + 1)[0].id));
+        changePlayer();
+    } else if (usedMoves.includes(parseInt(square.id))) {
+        status.textContent = `INVALID MOVE`;
+        setTimeout(() => status.textContent = `${currentPlayer.name}'s Turn`, 1000)
+    }
+}
+
+function changePlayer() {
+    isWinner();
+    if (status.textContent === `${currentPlayer.name} WINS!!!`) {
+        clock.innerText = "Clock"
+    }
+    else if (currentPlayer.name === playerX.name && onePlayer.checked == true) {
+        currentPlayer = playerO;;
+        status.textContent = `${currentPlayer.name}'s Turn`
+        randomSelect();
+    } else if (currentPlayer.name === playerX.name) {
+        currentPlayer = playerO;
+        status.textContent = `${currentPlayer.name}'s Turn`
+    } else if (currentPlayer.name === playerO.name) {
+        currentPlayer = playerX;
+        status.textContent = `${currentPlayer.name}'s Turn`
+    } isWinner();
+}
+
+
+function isWinner() {
+    for (win of wins) {
+        if (containsPlayerOMove(win).length === 3) {
+            endGame(playerO)
+        } else if (containsPlayerXMove(win).length === 3) {
+            endGame(playerX)
+        } else if (usedMoves.length === 9) {
+            draw();
+        }
+    }
+}
+
+function draw() {
+    status.textContent = `DRAW!`;
+    clearInterval(timer);
+    squares.forEach((square) => { square.removeEventListener('click', select) })
+    button.disabled = false;
+    count = 0;
+}
+
+function endGame(player) {
+    status.textContent = `${player.name} WINS!!!`;
+    squares.forEach((square) => { square.removeEventListener('click', select) })
+    button.disabled = false;
+    clearInterval(timer);
+    count = 0;
+}
+
+function countUp() {
+    count++
+    clock.innerText = count
+}
+
+function isBlockable() {
+    for (win of wins) {
+        if (containsPlayerXMove(win).length === 2 && containsPlayerOMove(win).length === 0) {
+            return win;
+        }
+    } return false;
+}
+function containsPlayerXMove(arr) {
+    let retArr = [];
+    for (move of playerX.moves) {
+        if (arr.includes(move)) {
+            retArr.push(move);
+        }
+    } return retArr;
+}
+function containsPlayerOMove(arr) {
+    let retArr = [];
+    for (move of playerO.moves) {
+        if (arr.includes(move)) {
+            retArr.push(move);
+        }
+    } return retArr; R
+}
```

Comment:
It seems like you could do this more simply by just directly querieng the `playerX` or `playerO` `move` property directly here. You could probably make this player agnostic either by checking `currentPlayer` moves or passing in the player as an additional argument.